test/distributed/test_c10d_nccl.py

```diff
@@ -3714,6 +3714,28 @@ def test_short(self, timing_enabled):
         else:
             self.assertTrue("duration_ms" not in last)
 
+    @requires_nccl()
+    @skip_but_pass_in_sandcastle_if(not TEST_MULTIGPU, "NCCL test requires 2+ GPUs")
+    def test_short_w_destroy_process_group(self):
+        if self.rank == self.MAIN_PROCESS_RANK:
+            return
+        pg = self._create_process_group_nccl()
+        device = self.local_device
+        a = torch.full((3, 4), float(self.rank), device=device)
+        for i in range(2):
+            f = pg.allreduce(a)
+        f.wait()
+        torch.cuda.synchronize(device=device)
+
+        # gah ok so now the duration_ms is populated best-effort since it can only happen outside "dump()" api
+        time.sleep(1)
+
+        t = pickle.loads(torch._C._distributed_c10d._dump_nccl_trace())
+        self.assertEqual(len(t), 2)
+        c10d.destroy_process_group()
```

Comment:
i think we could just add destroy call and second dump to the existing test_short.

but im not sure if its ok to call destroy twice, and i think the test infrastructure already calls destroy after each test?  double check that part